### DISCUSSION TITLE
Changing/adding the mp3 or background of a beatmap should not disturb ranked status/score submissions

### DISCUSSION THREAD

**Discussion by AadilRn** [ppy/osu#19799] [Q&A] [answered]
Currently, in osu!stable players are able to change the mp3 or background of their beatmaps without disturbing the ranked status of the beatmap and thus still able to submit scores. Some examples where players find this "stable feature" helpful are:

- Some players find some osu! backgrounds to be inappropriate/not to their liking and thus change the background to something different. (You may say the player can just turn the background dim to 100% but that does not affect the background in song select)
- Often times mappers will link a higher bitrate audio file of the song in the description for players to download and replace since oftentimes players will be dissatisfied with the audio quality of the song when the map was ranked.
- Some players will choose to replace the current mp3 in the beatmap with the same song sung by a different artist (a different cover). This is very prevalent with players who dislike ranked maps with nightcore/vocaloid covers of a song.

I understand how modifying the files can mess up the structure behind the game which is why I decided to just bring up this issue before lazer adopts pp submission because that is when I believe the flood gates will open and this github will be flooded by a thousand complaints, many of them being duplicates of this discussion.

I'm interested in hearing what the devs and other members of the community think should be done with this "feature".

**Comment by peppy** [comment, marked as answer]
> * Some players find some osu! backgrounds to be inappropriate/not to their liking and thus change the background to something different. (You may say the player can just turn the background dim to 100% but that does not affect the background in song select)

We will offer the ability to hide "explicit" background, and will probably give a way to hide all backgrounds from song select if required.

> * Often times mappers will link a higher bitrate audio file of the song in the description for players to download and replace since oftentimes players will be dissatisfied with the audio quality of the song when the map was ranked.

This is an unsupported workflow for now. I have no intention of supporting this unless there is a huge demand. It only worked in the past by chance.

> * Some players will choose to replace the current mp3 in the beatmap with the same song sung by a different artist (a different cover). This is very prevalent with players who dislike ranked maps with nightcore/vocaloid covers of a song.

Nope.

**Reply by AadilRn** [reply]
Interesting to see your views on some of the reasons for changing an mp3. Again I just wanted to bring your attention to this beforehand as I know you will most likely be spammed about this on Github and Twitter when pp submission becomes a thing. I am alright with marking your response as the answer.
